src/psyclone/psyGen.py

```diff
@@ -1199,26 +1213,49 @@ def parent(self, my_parent):
 
     @property
     def position(self):
+        '''
+        Find a Node's position relative to its parent Node (starting
+        with 0 if it does not have a parent).
+        :returns: relative position of a Node to its parent
+        :rtype: int
+        '''
         if self.parent is None:
-            return 0
+            return self.START_POSITION
         return self.parent.children.index(self)
 
     @property
     def abs_position(self):
-        ''' Find my position in the schedule. Needs to be computed
-            dynamically as my position may change. '''
-
+        '''
+        Find a Node's absolute position in the tree (starting with 0 if
+        it is the root). Needs to be computed dynamically from the
+        starting position (0) as its position may change.
+        :returns: absolute position of a Node in the tree
+        :raises InternalError: if the absolute position cannot be found
+        :rtype: int
+        '''
         if self.root == self:
-            return 0
-        found, position = self._find_position(self.root.children, 0)
+            return self.START_POSITION
+        found, position = self._find_position(self.root.children,
+                                              self.START_POSITION)
         if not found:
-            raise Exception("Error in search for my position in "
-                            "the tree")
+            raise InternalError("Error in search for Node position "
```

Comment:
This line is not covered.

Reply:
I had to modify the function `abs_position` in `class Node` to be able to reach that line. My idea was that search for a random Node that does not belong to the Schedule would trigger the exception. However, that did not happen because the condition `if self.root == self` did not check that the Node root is actually a Schedule instance (I guess that was implicitly assumed) and would return `START_POSITION`. I added the check that and then modified the `test_node_position` to test the constraint.